### ISSUE TITLE
AP of zero for all classes

### ISSUE THREAD

**Issue by martiningram** [kevinjliang/tf-Faster-RCNN#7]
Hi Kevin and Dan,

First of all, thanks for making this repo. It's the first implementation of Faster R-CNN for Tensorflow that looks really nice!

I am currently trying to get things to work on PASCAL VOC 2007. After black-listing the images you mentioned in https://github.com/kevinjliang/tf-Faster-RCNN/issues/2, training seems to run OK, and the loss goes down fine. However, the AP for all classes in validation and test seems to come out as `0.0`. As far as I can tell, my data is formatted correctly.

Do you have any guesses of what might be causing this?

Thanks,
Martin

**Comment by kevinjliang** [comment]
Hi Martin! Thanks for the comment! We're still doing testing on our end with PASCAL VOC ourselves, so this isn't the finalized version yet, but the AP should end up better than 0. How many epochs have you trained? In my experience, the APs typically start off as 0's across the board, but start becoming nonzero around epoch 4 or 5. Even with pre-trained weights to start, the RPN and Fast RCNN classifier layers are initialized at random, so it takes the model some time before it starts producing reasonable results.

**Comment by martiningram** [comment]
Thanks for the quick reply Kevin. This was after just one epoch (~2500 steps), so I am running for 10 epochs now to see whether that helps.

**Comment by martiningram** [comment]
Hi Kevin, by 10 epochs, the mAP has started to go up, though only to around 0.001 or so. If you don't mind me asking, what kind of mAP have you been able to achieve on PASCAL VOC 2007 so far?

**Comment by kevinjliang** [comment]
I'd run for at least 25 epochs. Also, how did you organize your data? Did you run the script [`PASCAL_VOC.py`](https://github.com/kevinjliang/tf-Faster-RCNN/blob/master/Data/scripts/PASCAL_VOC.py)? Because PASCAL VOC  is set up 25:25:50 for train:valid:test, we typically combine the default train and validation sets into training, and then split the original test into a new validation and test (so that it becomes 50:25:25).

Last time I ran, I ended up around 0.35 mAP on test. Still fiddling with some optimization settings.

**Comment by martiningram** [comment]
Hi Kevin, after 50 epochs I am getting around 11% mAP on the validation set using the original 25:25:50 PASCAL VOC split. I guess that's probably in line with the 35% you get with the 50:25:25 split, and I'll close this issue.

**Comment by Jasonsun1993** [comment]
@martiningram Hi Martin. Recently, I got a problem which was similar to yours about mAP and AP on my dataset. I used the LISA dataset to train this code. But the mAP and AP were always 0.0 after about 20 epochs. It really puzzled me a lot these days. Can you tell me how to solve this problem? Thanks a lot!
![5](https://user-images.githubusercontent.com/33592598/39312861-1e824016-49a3-11e8-8a6b-3f69c53bc271.jpg)

**Comment by paviddavid** [comment]
@Jasonsun1993 I got the same result with similar data (German Traffic Sign Detection Benchmark Dataset). Did you solve this issue ? Any ideas?
